adagrams/game.py

```diff
@@ -1,11 +1,168 @@
+            if word_count == letter_bank.count(letter):
+                continue
+            else:
+                return False
+        else:
+            return False
+    return True
+
 
+##-----------------------------------------##
+##-----------------WAVE 3------------------##
+##-----------------------------------------##
+"""
+3. Now you need a function returns the score of a given word as defined by the Adagrams game.
+-Has one parameter: word, which is a string of characters
+-Returns an integer representing the number of points
+-Each letter within word has a point value. The number of points of each letter is summed up to represent the total score of word
+-Each letter's point value is described in the table below
+-If the length of the word is 7, 8, 9, or 10, then the word gets an additional 8 points
+"""
 def score_word(word):
-    pass
+    score_chart = {"A":1, "E":1, "I":1, "O":1, "U":1, "L":1, "N":1, "R":1, "S":1, "T":1, "D":2,
+    "G":2, "B":3, "C":3, "M":3, "P":3, "F":4, "H":4, "V":4, "W":4, "Y":4, "K":5, "J":8, "X":8,
+    "Q":10, "Z":10
+    }
+    score = 0
+    if len(word) >=7:
+            score += 8
+    for letter in word:
+        score += score_chart[letter.upper()]
+            
+    return score
+
+
+
+##-----------------------------------------##
+##-----------------WAVE 4------------------##
+##-----------------------------------------##
+"""
+4. You need a way to find the highest scoring word. This function looks at the list of word_list and calculates which 
+    of these words has the highest score, applies any tie-breaking logic, and returns the winning word in a special 
+    data structure.
+-Has one parameter: word_list, which is a list of strings
+-Returns a tuple that represents the data of a winning word and it's score. The tuple must contain the following elements:
+    -index 0 ([0]): a string of a word
+    -index 1 ([1]): the score of that word
+-In the case of tie in scores, use these tie-breaking rules:
+    -prefer the word with the fewest letters...
+    -...unless one word has 10 letters. If the top score is tied between multiple words and one is 10 letters long, 
+        choose the one with 10 letters over the one with fewer tiles
+    -If the there are multiple words that are the same score and the same length, pick the first one in the supplied list
+    reference: https://www.geeksforgeeks.org/python-difference-between-sorted-and-sort/
+    https://www.geeksforgeeks.org/python-sort-list-according-second-element-sublist/
+"""
+def get_highest_word_score(word_list):    
+
+
+    # word, pos = max(word_list, key=highest_score)
+    # tuple_highest_word.append(word, pos)
+    # return tuple_highest_word
+# return max(word_list, key=get_value)
+# def get_value(word):
+#     """Return word letter score."""
+#     return sum(score_chart[letter] for letter in word)
+
+    score = []
+    highest_scores = []
+
+    for i in range(len(word_list)):
```

Comment:
In this case a for each loop would be a little more customary so we don't have to worry about accessing the indexes for each word. It would also avoid having to use that first if statement to check if the index is greater than 0.